pkg/piecestore/psclient/pieceranger.go

```diff
@@ -63,10 +63,36 @@ func (r *pieceRanger) Range(ctx context.Context, offset, length int64) (io.ReadC
 		return ioutil.NopCloser(bytes.NewReader([]byte{})), nil
 	}
 
+	// Making a copy, otherwise there will be a data race
+	// when another goroutine tries to write the cached size
+	// of this instance at the same time.
+	pbaCopy := &pb.PayerBandwidthAllocation{
```

Comment:
There's probably a shorter way of expressing this... otherwise since we're doing it more than once we might want to make a .Clone() pointer receiver

Reply:
👍

Reply:
I think there is `proto.Clone()` func but it needs to be checked if it work as expected